release/v1alpha1/types.go

```diff
@@ -0,0 +1,185 @@
+}
+
+// JobState the aggregate state of the job
+// Supported values include Pending, Failed, and Success.
+type JobState string
+
+const (
+	// JobStatePending not all job runs have completed
+	// Transitions to Failed or Success
+	JobStatePending JobState = "Pending"
+
+	// JobStateFailed failed job aggregation
+	JobStateFailed JobState = "Failed"
+
+	// JobStateSuccess successful job aggregation
+	JobStateSuccess JobState = "Success"
+)
+
+// JobStatus encapsulates the name of the job, all the results of the jobs, and an aggregated
+// result of all the jobs
+type JobStatus struct {
+	// jobName is the name of the job
+	JobName string `json:"name,omitempty"`
+
+	// MaxRetries maximum times to retry a job
+	MaxRetries int `json:"maxRetries,omitempty"`
+
+	// AggregateState is the overall success/failure of all the executed jobs
+	AggregateState JobState `json:"state,omitempty"`
+
+	// JobRunResults contains the links for individual jobs
+	JobRunResults []JobRunResult `json:"results,omitempty"`
+}
+
+// JobRunState the status of a job
+type JobRunState string
+
+const (
+	// JobRunStateTriggered job has been created but not scheduled
+	JobRunStateTriggered JobRunState = "Triggered"
+
+	// JobRunStatePending job is running and awaiting completion
+	JobRunStatePending JobRunState = "Pending"
+
+	// JobRunStateFailure job completed with errors
+	JobRunStateFailure JobRunState = "Failure"
+
+	// JobRunStateSuccess job completed without errors
+	JobRunStateSuccess JobRunState = "Success"
+
+	// JobRunStateAborted job was terminated early
+	JobRunStateAborted JobRunState = "Aborted"
+
+	// JobRunStateError job could not be scheduled
+	JobRunStateError JobRunState = "Error"
+)
+
+// JobRunResult the results of a prowjob run
+// The release-controller creates ProwJobs (prowv1.ProwJob) during the sync_ready control loop and relies on an informer
+// to process jobs, that it created, as they are completed. The JobRunResults will be created, by the release-controller
+// during the sync_ready loop and updated whenever any changes, to the respective job is received by the informer.
+type JobRunResult struct {
+	// Name unique name for the job run
+	Name string `json:"name,omitempty"`
+
+	// Namespace location where the job ran
+	Namespace string `json:"namespace,omitempty"`
+
+	// Cluster is which Kubernetes cluster is used to run the job
+	Cluster string `json:"cluster,omitempty"`
+
+	// RunID the unique identifier of the job
+	RunId int `json:"runId"`
+
+	// StartTime timestamp for when the job was created
+	StartTime metav1.Time `json:"startTime,omitempty"`
+
+	// CompletionTime timestamp for when the job goes into a final state
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// State the current state of the job run
+	State JobRunState `json:"state"`
+
+	// HumanProwResultsURL the html link to the prow results
+	HumanProwResultsURL string `json:"humanProwResultsURL"`
+
+	// Bucket the bucket artifacts have been upload into
```

Comment:
this needs clarification.  What's a bucket, where can I find it, how can I access it?

Reply:
My hope here was for an easy way to get to the GCS Bucket that contains the various logs for the ProwJob.  Unfortunately, it's only the name of the bucket.  We could piece it together, but it'll be fragile if/when `test-infra` changes.